api/v1alpha1/adminnetworkpolicy_types.go

```diff
@@ -0,0 +1,290 @@
+/*
+Copyright 2022.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// AdminNetworkPolicy is the Schema for the adminnetworkpolicies API
+type AdminNetworkPolicy struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of AdminNetworkPolicy.
+	// +optional
```

Comment:
Is the spec really optional?

Reply:
For [NetworkPolicy](https://github.com/kubernetes/api/blob/master/networking/v1/types.go#L37) and a few other types in the core API it is, but I agree logically it doesn't make sense